### DISCUSSION TITLE
Question: Is there an operator for `zip`?

### DISCUSSION THREAD

**Discussion by amadeu01** [pointfreeco/swift-prelude#160] [Q&A] [answered]
There is a `Zip` function in Overture
https://github.com/pointfreeco/swift-overture#zip-and-zipwith

However, I would like to know whether exists an operator for it

**Comment by stephencelis** [comment, marked as answer]
Hi @amadeu01! The functionality of `zip` is typically achieved using `curry` alongside `pure`, and `<*>` (apply).

``` swift
zip(with: User.init)(optionalId, optionalName, optionalEmail)

// vs.

pure(curry(User.init)) <*> optionalId <*> optionalName <*> optionalEmail
```

A shortcut to using `pure` is to sometimes use the `map` infix operator (`<$>` in Haskell, `<¢>` in this library):

``` swift
curry(User.init) <¢> optionalId <*> optionalName <*> optionalEmail
```
